core/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -1460,11 +1460,16 @@ public static double pow(final double a, final double b) {
      * Returns the distance between two GeoPointData locations, in meters, given objects to unpack.
      * Ignores altitude and accuracy.
      * Note that the arguments can be strings.
+     * Returns -1 if one of the arguments is null or the empty string.
      */
     public static Double distance(Object from, Object to) {
         String unpackedFrom = (String) unpack(from);
         String unpackedTo = (String) unpack(to);
 
+        if (unpackedFrom == null || "".equals(unpackedFrom) || unpackedTo == null || "".equals(unpackedTo)) {
+            return -1.0;

```

Comment:
You need to explicitly return a Double object of value -1, since J2ME doesn't auto-box. Build just :fire: :boom: :bomb: :boom: :fire: